File: app/src/main/java/com/mona/moviesapp/popular_people/PopularListScreen/controller/ListController.java
package com.mona.moviesapp.popular_people.PopularListScreen.controller;

import com.mona.moviesapp.popular_people.PopularListScreen.model.ListModel;
import com.mona.moviesapp.popular_people.PopularListScreen.view.PopularListActivity;
import com.mona.moviesapp.popular_people.pojo.PopularInfo;

public class ListController {

    private PopularListActivity popularListActivity;
    private ListModel listModel = new ListModel(this);

    String img_path = "https://image.tmdb.org/t/p/w500/";

    public ListController(PopularListActivity popularListActivity) {
        this.popularListActivity = popularListActivity;
        this.listModel = listModel;
    }

    public void changeAdapter(){
        popularListActivity.changeList();
    }

    public void getPopularList(PopularInfo popularInfo){
        popularListActivity.addPopularList(popularInfo);
    }

    public void callJson(int pagenum){
        String popularUrl= "https://api.themoviedb.org/3/person/popular?api_key=bd9eb9f62e484b7b3de4718afb6cd421&page="+pagenum;
        listModel.geturl(popularUrl);
    }

    public void searchingCall(String searchStr){
        String searchUrl="https://api.themoviedb.org/3/search/person?api_key=bd9eb9f62e484b7b3de4718afb6cd421&query="+searchStr;
        listModel.geturl(searchUrl);
    }
}

File: app/src/main/java/com/mona/moviesapp/popular_people/PopularListScreen/controller/ListController.kt
package com.mona.moviesapp.popular_people.PopularListScreen.controller

import com.mona.moviesapp.popular_people.PopularListScreen.model.ListModel
import com.mona.moviesapp.popular_people.PopularListScreen.view.PopularListActivity
import com.mona.moviesapp.popular_people.pojo.PopularInfo

class ListController(private val popularListActivity: PopularListActivity) {
    private var listModel = ListModel(this)

    internal var img_path = "https://image.tmdb.org/t/p/w500/"

    init {
        this.listModel = listModel
    }

    fun changeAdapter() {
        popularListActivity.changeList()
    }

    fun getPopularList(popularInfo: PopularInfo) {
        popularListActivity.addPopularList(popularInfo)
    }

    fun callJson(pagenum: Int) {
        val popularUrl = "https://api.themoviedb.org/3/person/popular?api_key=bd9eb9f62e484b7b3de4718afb6cd421&page=$pagenum"
        listModel.geturl(popularUrl)
    }

    fun searchingCall(searchStr: String) {
        val searchUrl = "https://api.themoviedb.org/3/search/person?api_key=bd9eb9f62e484b7b3de4718afb6cd421&query=$searchStr"
        listModel.geturl(searchUrl)
    }
}
